go/libkb/secret_store_secretservice.go

```diff
@@ -58,8 +62,28 @@ func (s *SecretStoreSecretService) maybeRetrieveSingleItem(mctx MetaContext, srv
 	return &item, nil
 }
 
-func (s *SecretStoreSecretService) RetrieveSecret(mctx MetaContext, username NormalizedUsername) (secret LKSecFullSecret, err error) {
-	defer mctx.TraceTimed("SecretStoreSecretService.RetrieveSecret", func() error { return err })()
+func (s *SecretStoreRevokableSecretService) ekstoreDir(mctx MetaContext, username string) string {
+	return fmt.Sprintf("ring/%s", username)
+}
+
+func (s *SecretStoreRevokableSecretService) secretlessEKStore(mctx MetaContext, username string) SecretlessErasableKVStore {
+	return NewSecretlessFileErasableKVStore(mctx, s.ekstoreDir(mctx, username))
+}
+
+func (s *SecretStoreRevokableSecretService) ekstore(mctx MetaContext, username string, keyringSecret []byte) ErasableKVStore {
+	keygen := func(mctx MetaContext, noise NoiseBytes) (xs [32]byte, err error) {
+		h := hkdf.New(sha256.New, append(noise[:], keyringSecret...), nil, []byte("Keybase-Derived-LKS-SecretBox-1"))
```

Comment:
can you move the context string to https://github.com/keybase/client/blob/master/go/libkb/constants.go#L602